lib/internal/readline/interface.js

```diff
@@ -150,7 +151,10 @@ function InterfaceConstructor(input, output, completer, terminal) {
   let prompt = '> ';
   let signal;
 
-  if (input?.input) {
+  if (input instanceof FileHandle) {
+    input = input.createReadStream();
+    crlfDelay = Infinity;
```

Comment:
Are we sure that this is the only option that needs to be set with a value different from the default while reading lines from a file?
Also, why isn't Infinity the default value of crlfDelay in all cases?

Reply:
>  I don't think adding an unnecessary breaking change is worth it

I thought doing that would fix the weird problems you were mentioning in https://github.com/nodejs/node/pull/42590#issuecomment-1114002784 instead of introducing unnecessary breakage. (would be interesting to know what the weird problem is though)

Reply:
The unnecessary breakage is to change the default. See the docs for more context: https://github.com/nodejs/node/blob/6afd3fcf653ed92063bafefa83661a076d241585/doc/api/readline.md#L998-L1003

Reply:
Is `\r\n` supposed to be considered as a single line even on posix? I thought line endings were `\r` on posix and `\r\n` on windows.

Reply:
`\r` and `\n` are two different characters, `\n` is the one that marks the end of the line on non-Windows systems, and the combination of both on Windows. `readline` will treat both as a line separator, unless there next to one another.

https://github.com/nodejs/node/blob/d36b60e69a00b60b8f2c260b8c376c9a16a152df/lib/internal/readline/interface.js#L1261-L1275